lib/std/mem.zig

```diff
@@ -947,6 +947,41 @@ test "len" {
     try testing.expect(len(c_ptr) == 2);
 }
 
+pub const LexicographicalOrder = enum(i2) {
+    lt = -1,
+    eq = 0,
+    gt = 1,
+
+    pub fn toMathOrder(self: LexicographicalOrder) math.Order {
+        return switch (self) {
+            .lt => .lt,
+            .eq => .eq,
+            .gt => .gt,
+        };
+    }
+};
+
+/// Returns `LexicographicalOrder.lt` if `a` appears before `b` in lexicographical order.
+/// Returns `LexicographicalOrder.gt` if `a` appears after `b` in lexicographical order.
+/// Returns `LexicographicalOrder.eq` if `a` and `b` are equal.
+/// See also `std.mem.order`.
+pub fn stringCompareZ(a: [*:0]const u8, b: [*:0]const u8) LexicographicalOrder {
```

Comment:
Please avoid the word "string" in mem functions. I don't want anyone to get the impression that this comparison is unicode-aware.

Inside `std.mem`, it is already implied that you are talking about consecutive memory regions. No need for the word "string" or any similar word.

Also, please use `std.math.Order`. The only reason for `LexicographicalOrder` to exist is the C ABI bleeding into the zig std lib, which I don't want to happen.

Reply:
Do you mean something like `std.mem.orderZ`?